GoRogue/MapGeneration/Region.cs

```diff
@@ -9,17 +10,56 @@
 
 namespace GoRogue.MapGeneration
 {
+    /// <summary>
+    /// Event fired when a region's Area is changed.
+    /// </summary>
+    [PublicAPI]
+    public class RegionAreaChanged : EventArgs
+    {
+        /// <summary>
+        /// The region whose area is changing
+        /// </summary>
+        public readonly Region Item;
+
+        /// <summary>
+        /// The previous Area
+        /// </summary>
+        public readonly Area Old;
+
+        /// <summary>
+        /// The Area to which we update
+        /// </summary>
+        public readonly Area New;
+
+        /// <summary>
+        /// The Event Arguments for when a region's area is changed
+        /// </summary>
+        /// <param name="item">The region whose area is changing</param>
+        /// <param name="oldArea">The former value of the Area</param>
+        /// <param name="newArea">The new value of the Area</param>
+        public RegionAreaChanged(Region item, Area oldArea, Area newArea)
+        {
+            Item = item;
+            Old = oldArea;
+            New = newArea;
+        }
+    }
     /// <summary>
     /// A region of the map with four sides of arbitrary shape and size
     /// </summary>
     [PublicAPI]
-    public class Region : IObjectWithComponents, IReadOnlyArea
+    public class Region : IObjectWithComponents
     {
         /// <summary>
         /// The Area of this region
         /// </summary>
         public PolygonArea Area { get; set; }
```

Comment:
The implementation of this property should fire the corresponding changed event as appropriate:
```suggestion
        /// <summary>
        /// The Area of this region
        /// </summary>
        public PolygonArea Area
        {
            get => _area;
            set
            {
                if (value == _area) return;
                
                _oldValue = _area;
                _area = value;
                AreaChanged?.Invoke(this, new RegionAreaChangedEventArgs(this, _oldValue, value));
            }
        }
        private PolygonArea _area;
```